python/lib/communication/dmod/communication/client.py

```diff
@@ -117,133 +161,266 @@ async def async_recv(self) -> str:
 
     @property
     @abstractmethod
-    def client_ssl_context(self) -> ssl.SSLContext:
+    def endpoint_uri(self) -> str:
         """
-        Get the client SSL context property.
+        The endpoint for the client to connect to when opening a connection.
 
         Returns
         -------
-        ssl.SSLContext
-            The client SSL context for secure connections.
+        str
+            The endpoint for the client to connect to when opening a connection.
         """
         pass
 
+    @property
+    def client_ssl_context(self) -> Optional[ssl.SSLContext]:
```

Comment:
Does this _need_ to be public?

Reply:
I don't think so, no.  I went ahead and changed that as well.